internal/controller/ols_app_redis_assets.go

```diff
@@ -70,28 +127,71 @@ func (r *OLSConfigReconciler) generateRedisDeployment(cr *olsv1alpha1.OLSConfig)
 	return &deployment, nil
 }
 
-func (r *OLSConfigReconciler) generateRedisService(cr *olsv1alpha1.OLSConfig) (*corev1.Service, error) {
-	DeploymentSelectorLabels := map[string]string{
-		"app.kubernetes.io/component":  "redis-server",
-		"app.kubernetes.io/managed-by": "lightspeed-operator",
-		"app.kubernetes.io/name":       "lightspeed-service-redis",
-		"app.kubernetes.io/part-of":    "openshift-lightspeed",
+// updateRedisDeployment updates the deployment based on CustomResource configuration.
+func (r *OLSConfigReconciler) updateRedisDeployment(ctx context.Context, existingDeployment, desiredDeployment *appsv1.Deployment) error {
+	changed := false
+
+	// Validate deployment annotations.
+	if existingDeployment.Annotations == nil ||
+		existingDeployment.Annotations[OLSConfigHashKey] != r.stateCache[OLSConfigHashStateCacheKey] {
+		updateDeploymentAnnotations(existingDeployment, map[string]string{
+			OLSConfigHashKey: r.stateCache[OLSConfigHashStateCacheKey],
+		})
+		// update the deployment template annotation triggers the rolling update
+		updateDeploymentTemplateAnnotations(existingDeployment, map[string]string{
+			OLSConfigHashKey: r.stateCache[OLSConfigHashStateCacheKey],
+		})
+
+		changed = true
 	}
+
+	// Validate volume mounts for a specific container in deployment.
+	if commandChanged, err := setCommand(existingDeployment, desiredDeployment.Spec.Template.Spec.Containers[0].Command, OLSAppRedisDeploymentName); err != nil {
+		return err
+	} else if commandChanged {
+		changed = true
+	}
+
+	if changed {
+		r.logger.Info("updating OLS redis deployment", "name", existingDeployment.Name)
+		if err := r.Update(ctx, existingDeployment); err != nil {
+			return err
+		}
+	} else {
+		r.logger.Info("OLS redis deployment reconciliation skipped", "deployment", existingDeployment.Name, "olsconfig hash", existingDeployment.Annotations[OLSConfigHashKey])
+	}
+
+	return nil
+}
+
+func (r *OLSConfigReconciler) generateRedisService(cr *olsv1alpha1.OLSConfig) (*corev1.Service, error) {
+	internalTrafficPolicy := corev1.ServiceInternalTrafficPolicyCluster
+	ipFamilies := []corev1.IPFamily{corev1.IPv4Protocol}
+	ipFamilyPolicy := corev1.IPFamilyPolicySingleStack
 	service := corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      OLSAppRedisServiceName,
 			Namespace: cr.Namespace,
 			Labels:    DeploymentSelectorLabels,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/serving-cert-secret-name": OLSAppRedisCertsSecretName,
+			},
 		},
 		Spec: corev1.ServiceSpec{
+			InternalTrafficPolicy: &internalTrafficPolicy,
+			IPFamilies:            ipFamilies,
+			IPFamilyPolicy:        &ipFamilyPolicy,
 			Ports: []corev1.ServicePort{
 				{
 					Port:       OLSAppRedisServicePort,
+					Protocol:   corev1.ProtocolTCP,
 					Name:       "server",
 					TargetPort: intstr.Parse("server"),
 				},
 			},
-			Selector: DeploymentSelectorLabels,
+			Selector:        DeploymentSelectorLabels,
```

Comment:
per [previous comments](https://github.com/openshift/lightspeed-operator/pull/25/files#diff-9a71d901f9ea521f81e45a15ebc88103ed9a382cb65b5942e24d20946818474cR20),  here is to be replaced by the new function call.

Reply:
Done, Changed as suggested.